lib/fog/vsphere/models/compute/snapshot.rb

```diff
@@ -0,0 +1,42 @@
+require 'fog/compute/models/server'
+
+module Fog
+  module Compute
+    class Vsphere
+      class Snapshot < Fog::Model
+
+        identity  :ref
+        attribute :server_id
+
+        attribute :name
+        attribute :quiescedi, :default => false
+        attribute :description, :default => ''
+        attribute :create_time
+        attribute :power_state, :default => 'none'
+        attribute :ref
+        attribute :mo_ref
+        attribute :tree_node
+        attribute :snapshot_name_chain
+        attribute :ref_chain
+
+        attr_reader :service
+
+        def initialize(attributes = {})
+          super attributes
+        end
+
+        def child_snapshots(filters = {})
+          requires :mo_ref
+          service.snapshots(
+            { :server_id => server_id, :parent_snapshot => self }.update(filters)
+          )
+        end
+
+        def get(snapshot_ref)
```

Comment:
Hi @plribeiro3000 I am a colleague of @ireznice.

(do you mean `Fog::Compute::Vsphere::Real === connection`? e.g. `connection.snapshots(server_id: ...).get(snapshot_ref)`)

This method recursively goes through snapshot tree and finds/gets the snapshot by its `ref` (also self may be returned). Unfortunately it has to be done recursively by "rbvmomi" call for each snapshot node in tree.

In current implementation, `Snapshots#get` recursively calls `get` on `Snapshot`. It could be refactored in `Snapshots#get` but I think current version is slightly more readable.

Do you suggest to rewrite it to `Snapshosts`? We will fully accept your opinion.